src/assemble_workflow/bundle_rpm.py

```diff
@@ -89,14 +92,27 @@ def build(self, name: str, dest: str, archive_path: str) -> None:
         shutil.move(f"{min_bin_env_path}.backup", min_bin_env_path)
         shutil.move(min_dest_path, min_source_path)
 
+        # Append correct value substitute to the spec file
+        architecture_alt_rpm = {
+            "x64": "x86_64",
+            "arm64": "aarch64",
+        }
+        replaces = ('##VERSION##', build_kls.version), ('##ARCHITECTURE_ALT##', architecture_alt_rpm[build_kls.architecture])
+
+        with open(os.path.join(ext_dest, f"{self.filename}.rpm.spec"), 'r') as fp:
+            spec_lines = fp.read()
+
+        with open(os.path.join(ext_dest, f"{self.filename}.rpm.final.spec"), 'w') as fp:
+            fp.write(reduce(lambda a, key_value: a.replace(*key_value), replaces, spec_lines))
```

Comment:
If you're going to be doing template replacement extract that into at least a method, or ideally into a separate class, or ideally use a built-in way to edit templates in Python, https://stackabuse.com/formatting-strings-with-the-python-template-class/

Reply:
Completely removed this section right now.